File: walkway/walkway.cpp
#include <bits/stdc++.h>

using namespace std;


int main(){
	
	freopen("input.txt", "r", stdin);
	freopen("output.txt", "w", stdout);
	
	int N, K;
	cin >> N >> K;
	
	vector<int> H;
	H.resize(N);
	for (int i = 0; i < N; i++){
		cin >> H[i];
	}
	
	
	sort(H.begin(), H.end());
	
	
	int result = 1000000;
	int temp = 0;
	for (int i = 0; i < N - K + 1; i++){
		temp = abs(H[i + K - 1] - H[i]);
		
		result = min(result, temp);
	}
	
	cout << result << "\n";
	
}

File: walkway/walkway.py

N, K = list(map(int, input().strip().split()))


H = list(map(int, input().strip().split()))

H.sort()

#print(H)

for i in range(N - 1):
    H[i] = abs(H[i] - H[i + 1])
H[N - 1] = 0

#print(H)

#print()

result = 1000000

for i in range(N - K + 1):
    #temp = H[i:i+K-1]
    #print(temp)
    somma = sum(H[i:i+K-1])
    result = min(result, somma)


print(result)

